### DISCUSSION TITLE
delete statement generates invalid SQL starting with 2.0.21

### DISCUSSION THREAD

**Discussion by jfowler2** [sqlalchemy/sqlalchemy#11624] [Usage Questions]
We have encountered what appears to be a bug with a delete statement.  Essentially we have a parent class with two child classes, and we issue a statement like the following:

`delete(Parent).where(Parent.id == 1)`

With 2.0.20 and earlier, this would generate SQL like the following:

`DELETE FROM parent WHERE parent.id = :id_1`

With 2.0.21 and later, it generates SQL like the following:

`DELETE FROM parent LEFT OUTER JOIN child_a ON parent.id = child_a.parent_id LEFT OUTER JOIN child_b ON parent.id = child_b.parent_id , child_a, child_b WHERE parent.id = :id_1`

which is invalid because delete statements can't have left outer joins.  

Here's an example script illustrating the issue.  It doesn't connect to an actual DB, just prints out the generated SQL to illustrate the issue.  

```
import enum
from typing import Annotated

from sqlalchemy import ForeignKey, delete
from sqlalchemy.orm import Mapped, declarative_base, mapped_column

Base = declarative_base()

IntPK = Annotated[int, mapped_column(primary_key=True, autoincrement=True)]


@enum.unique
class Discriminator(enum.StrEnum):
    A = "a"
    B = "b"


class Parent(Base):
    id: Mapped[IntPK]
    discriminator: Mapped[Discriminator]

    __tablename__ = "parent"
    __mapper_args__ = {
        "polymorphic_identity": "parent",
        "with_polymorphic": "*",
        "polymorphic_on": "discriminator",
    }


class ChildA(Parent):
    parent_id: Mapped[int] = mapped_column(
        ForeignKey("parent.id"), primary_key=True
    )

    __tablename__ = "child_a"
    __mapper_args__ = {
        "polymorphic_identity": "a",
    }


class ChildB(Parent):
    parent_id: Mapped[int] = mapped_column(
        ForeignKey("parent.id"), primary_key=True
    )

    __tablename__ = "child_b"
    __mapper_args__ = {
        "polymorphic_identity": "b",
    }


stmt = delete(Parent).where(Parent.id == 1)
print(stmt)
```

**Comment by zzzeek** [comment]
this is #11625 thanks for reporting
